nondeterministic failure of RelativeIso8601DateTest.testLazyInstance

I had a build fail due to this:

```
   [testng] FAILED: testLazyInstance
   [testng] java.lang.AssertionError: expected [1.46099929E12] but found [1.46099942E12]
   [testng] 	at org.testng.Assert.fail(Assert.java:94)
   [testng] 	at org.testng.Assert.failNotEquals(Assert.java:513)
   [testng] 	at org.testng.Assert.assertEquals(Assert.java:260)
   [testng] 	at org.testng.Assert.assertEquals(Assert.java:273)
   [testng] 	at htsjdk.samtools.util.RelativeIso8601DateTest.testLazyInstance(RelativeIso8601DateTest.java:19)
   [testng] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   [testng] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   [testng] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [testng] 	at java.lang.reflect.Method.invoke(Method.java:497)
   [testng] 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
   [testng] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
   [testng] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
   [testng] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
   [testng] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
   [testng] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
   [testng] 	at org.testng.TestRunner.privateRun(TestRunner.java:782)
   [testng] 	at org.testng.TestRunner.run(TestRunner.java:632)
   [testng] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
   [testng] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
   [testng] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
   [testng] 	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
   [testng] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
   [testng] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
   [testng] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
   [testng] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
   [testng] 	at org.testng.TestNG.run(TestNG.java:1064)
   [testng] 	at org.testng.TestNG.privateMain(TestNG.java:1385)
   [testng] 	at org.testng.TestNG.main(TestNG.java:1354)
```

--- a/src/test/java/htsjdk/samtools/util/RelativeIso8601DateTest.java
+++ b/src/test/java/htsjdk/samtools/util/RelativeIso8601DateTest.java
@@ -10,13 +10,17 @@ import java.util.List;
 /** @author mccowan */
 
 public class RelativeIso8601DateTest {
+
+    // 1 second resolution is ISO date
+    private final static double DELTA_FOR_TIME = 1000;
+
     @Test
     public void testLazyInstance() {
         final RelativeIso8601Date lazy = RelativeIso8601Date.generateLazyNowInstance();
         Assert.assertEquals(lazy.toString(), RelativeIso8601Date.LAZY_NOW_LABEL);
         Assert.assertEquals(lazy.toString(), RelativeIso8601Date.LAZY_NOW_LABEL);
         Assert.assertEquals(lazy.toString(), RelativeIso8601Date.LAZY_NOW_LABEL);
-        Assert.assertEquals(lazy.getTime(), new Iso8601Date(new Date(System.currentTimeMillis())).getTime(), 1000); // 1 second resolution is ISO date
+        Assert.assertEquals(lazy.getTime(), new Iso8601Date(new Date(System.currentTimeMillis())).getTime(), DELTA_FOR_TIME);
         // Assert no exception thrown; this should be valid, because toString should now return an iso-looking date.
         new Iso8601Date(lazy.toString());
     }
@@ -33,7 +37,7 @@ public class RelativeIso8601DateTest {
 
         for (final RelativeIso8601Date nonLazy : testDates) {
             Assert.assertFalse(nonLazy.toString().equals(RelativeIso8601Date.LAZY_NOW_LABEL));
-            Assert.assertEquals((double) nonLazy.getTime(), (double) time);
+            Assert.assertEquals((double) nonLazy.getTime(), (double) time, DELTA_FOR_TIME);
             // Assert no exception thrown; this should be valid, because toString return an iso-looking date.
             new RelativeIso8601Date(nonLazy.toString());
         }
@@ -44,6 +48,6 @@ public class RelativeIso8601DateTest {
         final String s = new Iso8601Date(new Date(12345)).toString();
         final Iso8601Date iso8601Date = new Iso8601Date(s);
         final RelativeIso8601Date relativeIso8601Date = new RelativeIso8601Date(s);
-        Assert.assertEquals(relativeIso8601Date.getTime(), iso8601Date.getTime());
+        Assert.assertEquals(relativeIso8601Date.getTime(), iso8601Date.getTime(), DELTA_FOR_TIME);
     }
 }

